tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+  - `CrossLandDefId` The cross language definitionId for the node.
+- `Properties (p)` : Use this for other properties of the node. The currently supported tags
+  - `SubKind` Similar to kind, use this to make the node more specific. e.g.   `Kind = 'Type'`, and `SubKind = 'class'` or somethign specific to you language. We also use this to make the navigation icon it will overide kind.
+  - `IconName` Use this only if you are looking to add a custom icon different from language wide defaults.
+- `TopTokens (tt)` : The main data of the node. This is all the tokens that actually define the node. e.g. For a class this would include the access modifier, the class name, any attributes or decorators e.t.c. For a method this would include the return type, method name, parameters e.t.c. See StructureTokesn description below for more info.
+- `BottomToken (bt)` : Data that closes out the node. Depending on the language this would include the closing curly brace and/or empty lines. You can simulate an empty line by adding an empty token (content token with no value) and a lineBreak token.
+- `Children (c)` : The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods.
+
+### StructuredToken
+- `Value (v)` : The token value which will be dispalyed. Can be null.  Spacing tokens don't need to have value.
+- `Id (i)` : This is essentially whatever existing parser was asigning to the token DefinitionId. You dont have to assign a tokenId.
+- `Kind (k)` : An enum
+  - `Content` Specifies that the token is content
+  - `LineBreak` Space token indicating switch to new line.
+  - `NoneBreakingSpace` Regular single space
+  - `TabSpace` 4 NoneBreakingSpaces
+  - `ParameterSeparator` Use this between method parameters. Depending on user setting this would result in a singlespace or new line
+  - `Url` A url token should have `LinkText` property and the url/link should be the token value.
+  All tokens should be content except for spacing tokens and url. ParameterSeparator should be used between method or function parameters. 
+- `Tags (t)` : Use this for opt in or opt out boolean properties.The currently supported tags are 
+  - `SkippDiff`  Indicate that a token should not be used in computatation of diff.
+  - `Deprecated` Mark a token as deprecated
+- `Properties (p)` : Properties of the token.
+  - `GroupId` : `doc` to group consecutive comment tokens as documentation.
+  - `NavigateToId` Id for navigating to where the node where the token is defined. Should match the definitionId of the node.
+- `RenderClasses (rc)` : Add css classes for how the tokens will be rendred. Classes currently being used are `text` `keyword` `punc` `tname` `mname` `literal` `sliteral` `comment` Feel free to add your own custom class. Whatever custom classes you use please provide us the appriopriate css for the class so we can update APIView.
```

Comment:
Again, there should be an enum (extensible enum perhaps) with the supported values. 

"Feel free to add your own custom class. Whatever custom classes you use please provide us the appriopriate css for the class so we can update APIView."

This should just be in a special topics section about what render classes are, how they work, etc.